site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md

```diff
@@ -0,0 +1,403 @@
+  separately, with no coordination between them. In this case, a user named "ryan" from one IDP is probably not
+  the same human as the user named "ryan" from the other IDP. The group named "admins" from one IDP may
+  have a totally different intended meaning then the group named "admins" from the other IDP.
+  In this case, it is not desirable for usernames and group names from one IDP to conflict with the usernames
+  and group names from the other IDP.
+- Imagine two IDPs which both use corporate email addresses as usernames. These email addresses are
+  assigned by IT and are not adjustable by the individual users. In this case, it may be desirable for a human
+  user to be able to authenticate using either IDP and end up being assigned the same username in
+  Kubernetes clusters either way. However, it may or may not be the case that the user's group names from one IDP
+  are meant to represent the same Kubernetes groups as the same group names from the other IDP.
+  In this case, it is not a concern for usernames from one IDP to conflict with the usernames from the other IDP,
+  however it still might be a concern for group names from one IDP to conflict with the group names from the other IDP.
+
+You can easily add configuration to the FederationDomain to handle username and group name conflicts.
+1. When it is not desirable for usernames to conflict, then a simple solution is to use identity transformations
+   to change all usernames to have a prefix which is unique to each IDP within that FederationDomain. For example,
+   usernames `ldap:ryan` and `gitlab:ryan` will be considered two different users by Kubernetes.
+2. Similarly, when it is not desirable for group names to conflict, then a simple solution is to use identity transformations
+   to change all group names to have a prefix which is unique to each IDP within that FederationDomain.
+
+Refer to the next section to learn about how to configure identity transformations to add prefixes to usernames
+and group names.
+
+## Identity transformations and policies
+
+When a user authenticates, the configuration of the OIDCIdentityProvider, ActiveDirectoryIdentityProvider, or 
+LDAPIdentityProvider resource determines how the user's username and group names will be extracted from the external
+identity provider in a protocol-specific way (e.g. via OIDC ID token claims or LDAP record attributes).
+
+Then, operating on the username and group names extracted from the external IDP:
+- Identity **transformations** can change either the user's username or group names.
+- Identity **policies** can reject the user's authentication based on their username and/or groups.
+
+Identity transformations and policies are configured on the FederationDomain, so they are specific to that specific
+FederationDomain's usage of the external identity provider.
+
+Transformations and policies are configured using the Common Expression Language (CEL) programming language.
```

Comment:
Shall we link to Kubernetes CEL docs? 
https://kubernetes.io/docs/reference/using-api/cel/.  Since CEL is relatively new we may want to emphasize how these are becoming standard in Kubernetes and not an arbitrary decision made by Pinniped.

Reply:
I don't think we need to defend our choices in the user guide. We did that in the proposal doc.